stdlib/urllib/parse.pyi

```diff
@@ -152,24 +153,24 @@ def unquote_plus(string: str, encoding: str = ..., errors: str = ...) -> str: ..
 @overload
 def urldefrag(url: str) -> DefragResult: ...
 @overload
-def urldefrag(url: bytes | None) -> DefragResultBytes: ...
+def urldefrag(url: bytes | bytearray | None) -> DefragResultBytes: ...
 def urlencode(
     query: Mapping[Any, Any] | Mapping[Any, Sequence[Any]] | Sequence[tuple[Any, Any]] | Sequence[tuple[Any, Sequence[Any]]],
     doseq: bool = ...,
-    safe: str | bytes = ...,
+    safe: str | ReadableBuffer = ...,
     encoding: str = ...,
     errors: str = ...,
-    quote_via: Callable[[AnyStr, str | bytes, str, str], str] = ...,
+    quote_via: Callable[[AnyStr, str | ReadableBuffer, str, str], str] = ...,
```

Comment:
This is dangerous because it makes the type more restrictive for callers: the callback now has to accept any ReadableBuffer. Looking at the code, it just gets passed whatever `safe` is, so we could make the function generic over a TypeVar bound to `str | Iterable[int]`, and use that for both `safe` and this argument to `quote_via`.

Reply:
👍

Reply:
This wasn't addressed, the callback still takes a very wide type.

Reply:
Done, sorry, I've missed this comment.